obs-browser-plugin.cpp

```diff
@@ -524,15 +524,19 @@ bool obs_module_load(void)
 		 * the default and on 2 or more adapters */
 		const wchar_t **device = blacklisted_devices;
 		while (*device) {
-			if (!!wstrstri(deviceId.c_str(), *device)) {
-				hwaccel = false;
-				blog(LOG_INFO, "[obs-browser]: "
-					       "Blacklisted device "
-					       "detected, "
-					       "disabling browser "
-					       "source hardware "
-					       "acceleration.");
-				break;
+			if (adapterCount >= 2 ||
+			    (adapterCount = 1 && !wstrstri(deviceId.c_str(),
+							   L"Intel"))) {
+				if (!!wstrstri(deviceId.c_str(), *device)) {
+					hwaccel = false;
+					blog(LOG_INFO, "[obs-browser]: "
+						       "Blacklisted device "
+						       "detected, "
+						       "disabling browser "
+						       "source hardware "
+						       "acceleration.");
+					break;
+				}
```

Comment:
The condition should probably be moved outside of the loop as it isn't influenced by the loop variables.

Reply:
done, thanks R1ch